alternatives.c

```diff
@@ -932,14 +932,15 @@ static int removeAll(const char * title, const char * altDir, const char * state
     struct alternativeSet set;
 
     int alt;
+    int ret_val = 0;
 
     if (readConfig(&set, title, altDir, stateDir, flags)) return 2;
 
     for (alt = 0; alt < set.numAlts; alt++) {
-        removeService(title, set.alts[alt].master.target, altDir, stateDir, flags);
+        ret_val += removeService(title, set.alts[alt].master.target, altDir, stateDir, flags);
```

Comment:
Hups, sorry I have missed the +. But anyway I would still go with the ternary approach.